src/app/app.pri

```diff
@@ -40,3 +40,4 @@ stacktrace {
 
 # upgrade code
 HEADERS += $$PWD/upgrade.h
+SOURCES += $$PWD/upgrade.cpp
```

Comment:
a bit unrelated to this PR, how about move this to the `SOURCES` above (and also `HEADERS`).

Reply:
Sure.
I don't know the initial reason for this. Maybe it was here because "upgrade" was a temporary solution. But now it pretty much is a stable part.